### PR TITLE
Setup webpack with rails

### PR DESCRIPTION
**Pull request by sajibarafatsiddiqui**
- Created a new Rails app called 'hello-rails-react'.
- Used the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verified that the jsbundling-rails gem is added to your Gemfile.
- Initialized project with Git.
- Installed needed node dependencies as described in the previous lesson's tutorial.
- Created a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Updated webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Made sure that your project has a Postgres database set up.
- Created a table for storing your messages and create 5 different greetings.
- Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Created a static view that will be the root of your app.
- Created the App component with react-router and render it in your static view.
- Created the Greeting component that will display a greeting. Set it up as a route in your App component.
- Created a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Displayed the random greeting in your Greeting component.

### CODE REVIEW

**Review by Reem-lab — COMMENTED** [review]
# Hi @sajibarafatsiddiqui, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a professional README suit the project information so kindly update your README with your project info and don't forget to remove the live demo section from your README since you don't have a live demo link to the project and it is not required for now 💯🚀✨🔥🔥

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by sajibarafatsiddiqui** [comment]
@Reem-lab why not you mention the reason? Why are you giving a default message? R u a robot?

**Comment by Reem-lab** [comment]
Hello @sajibarafatsiddiqui no I am not a robot I think it would be great if you have a professional README since it is important to have it so in the future the recruiters or any user could be able to see about the project and how to run it and more information about it ✨

**Comment by sajibarafatsiddiqui** [comment]
> Hello @sajibarafatsiddiqui no I am not a robot I think it would be great if you have a professional README since it is important to have it so in the future the recruiters or any user could be able to see about the project and how to run it and more information about it

@Reem-lab I am not getting you. I think you are watching readme file of dev branch not my working branch.

**Comment by Reem-lab** [comment]
Hello @sajibarafatsiddiqui no I was watching it on setupwebpackwithrails this branch and you were putting the budget app project inormation 👍🏻 but I think now you just updated it 



image

File: README.md [hunk 1]
```diff
@@ -1,2 +1,177 @@
...
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+> I would like to thank  <a href="https://www.behance.net/gregoirevella">Gregoire Vella on Behance</a> I have followed his 
+<a href="https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=&">design guildlines.</a> on behance.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (t-yanick) [inline README.md]: - [ ] Great job working on this milestone and updating the ReadMe. Unfortunately, when I click on the MIT license link, it gives me an error 404 page. It is not professional to allow broken links in our codebase. Kindly fix this issue.
image

**Comment by sajibarafatsiddiqui** [comment]
> Hello @sajibarafatsiddiqui no I was watching it on setupwebpackwithrails this branch and you were putting the budget app project inormation 👍🏻 but I think now you just updated it
> 
> image

The description was changed before sending to review . However I made changes. Now you are looking like human being. As you are pointing the objections objectively not throwing a default message.

**Comment by Reem-lab** [comment]
Hello @sajibarafatsiddiqui this is the commit after my review 💪🏻🔥🙏🏻

image

**Comment by sajibarafatsiddiqui** [comment]
> Hello @sajibarafatsiddiqui this is the commit after my review 💪🏻🔥🙏🏻
> 
> image

@Reem-lab Ok. It seems  forgot to save the readme file in vscode before committing. 🤦‍♂️. However greetings from me and my daughter. Thanks

File: app/javascript/application.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Route, Routes, BrowserRouter as Router } from 'react-router-dom';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import Greeting from './greeting';
+import store from './store';
+
+const App = () => (
```
Comment (t-yanick) [inline app/javascript/application.js:8]: - [ ] As per project requirements, you should rename this compnent from application.js to App.js
image

File: app/javascript/greetingSlice.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// redux/rockets/rocketsSlice.js
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseUrl = 'api/v1/random_greeting';
+// const apiKey = 'f7b89fe31c77772c4972eea6dc268fcb';
```
Comment (t-yanick) [inline app/javascript/greetingSlice.js]: - [ ] Kindly remove this commented line of code from your codebase. This keeps it clean and professional. You can always revisit old pieces of codes with git history

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Required gem and packages used
:heavy_check_mark: Route create
:heavy_check_mark: Migration and model present
:heavy_check_mark: Correct Gitflow used


There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (sajibarafatsiddiqui) [reply README.md]: The link is restored!!

File: app/javascript/greetingSlice.js (hunk 3 above)
Comment (sajibarafatsiddiqui) [reply app/javascript/greetingSlice.js]: The comment has been removed!!

File: app/javascript/application.js (hunk 2 above)
Comment (sajibarafatsiddiqui) [reply app/javascript/application.js:8]: @t-yanik I am astonished by your comment!!! The component name is App and from your image, it is clear also. Why are you messing the component name with the file name?🤦‍♂️

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @sajibarafatsiddiqui 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**